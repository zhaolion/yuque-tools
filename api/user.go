package api

import (
	"fmt"
	"time"
)

type UserCurrentResponse struct {
	Data UserDetail `json:"data"`
}

func (resp *UserCurrentResponse) String() string {
	return fmt.Sprintf("%+v", resp.Data)
}

type UserDetail struct {
	ID               int       `json:"id"`
	Type             string    `json:"type"`
	SpaceID          int       `json:"space_id"`
	AccountID        int       `json:"account_id"`
	Login            string    `json:"login"`
	Name             string    `json:"name"`
	AvatarURL        string    `json:"avatar_url"`
	BooksCount       int       `json:"books_count"`
	PublicBooksCount int       `json:"public_books_count"`
	FollowersCount   int       `json:"followers_count"`
	FollowingCount   int       `json:"following_count"`
	Public           int       `json:"public"`
	Description      string    `json:"description"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
	Serializer       string    `json:"_serializer"`
}

// UserCurrent 获取认证的用户的个人信息
// curl --location --request GET 'https://www.yuque.com/api/v2/user' \
//	--header 'X-Auth-Token: abcd'
func (c *Client) UserCurrent() (*UserCurrentResponse, error) {
	var (
		response UserCurrentResponse
		apiErr   YuqueError
	)
	resp, err := c.Client.R().
		SetError(&apiErr).
		SetResult(&response).
		Get("user")
	if err != nil {
		return nil, err
	}
	if resp.IsSuccess() {
		return &response, nil
	}

	return nil, &apiErr
}
