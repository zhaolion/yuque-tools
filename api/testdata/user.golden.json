{
  "data": {
    "id": 114999,
    "type": "User",
    "space_id": 0,
    "account_id": 20368,
    "login": "zhaolion",
    "name": "zhaolion",
    "avatar_url": "https://cdn.nlark.com/yuque/0/2020/png/114999/1604419869502-avatar/f203b97a-eaf1-4112-8211-919c8c68c326.png",
    "books_count": 4,
    "public_books_count": 3,
    "followers_count": 0,
    "following_count": 0,
    "public": 1,
    "description": "可爱的反派角色",
    "created_at": "2018-04-26T14:56:05.000Z",
    "updated_at": "2022-09-26T07:40:46.000Z",
    "_serializer": "v2.user_detail"
  }
}
